lib/network/modules/components/nodes/util/ShapeBase.js

```diff
@@ -82,17 +106,19 @@ class ShapeBase extends NodeBase {
    * @param {number} y
    */
   updateBoundingBox(x, y) {
+    // TODO: Same/similar code is at least 4 places!! 
     this.boundingBox.top = y - this.options.size;
     this.boundingBox.left = x - this.options.size;
     this.boundingBox.right = x + this.options.size;
     this.boundingBox.bottom = y + this.options.size;
 
     if (this.options.label !== undefined && this.labelModule.size.width > 0) {
+      this.boundingBox.top = Math.max(this.boundingBox.top, this.labelModule.size.top);
```

Comment:
You have `Math.max` here and `Math.max` at the bottom too. I think this one should be `Math.min` to include the label in the bounding box. I don't think this will solve all the issues though.